src/main/java/org/roaringbitmap/Container.java

```diff
@@ -99,6 +99,32 @@ else if (x instanceof BitmapContainer)
     }
 
     /**
+     * Computes the bitwise AND of this container with another
+     * (intersection). This container as well as the provided container are
+     * left unaffected.
+     *
+     * @param x other container
+     * @return aggregated container
+     */
+    public int andCardinality(Container x) {
+    	if (this.getCardinality() == 0)
+    		return 0;
+    	else if (x.getCardinality() ==0)
+    		return 0;
+		else
+		{
+    		if (x instanceof ArrayContainer)
+            	return and((ArrayContainer) x).getCardinality();

```

Comment:
`andCardinality(Container)` should be called here instead of `and(Container).getCardinality()`